packages/manager/apps/web/client/app/hosting/database/order/public/hosting-database-order-public.service.js

```diff
@@ -0,0 +1,105 @@
+import sortBy from 'lodash/sortBy';
+import toNumber from 'lodash/toNumber';
+
+import { PRODUCT_NAME } from './hosting-database-order-public.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiHostingWeb,
+    OvhApiMe,
+    OvhApiOrder,
+    OvhApiOrderCatalogPublic,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiHostingWeb = OvhApiHostingWeb;
+    this.OvhApiMe = OvhApiMe;
+    this.OvhApiOrder = OvhApiOrder;
+    this.OvhApiOrderCatalogPublic = OvhApiOrderCatalogPublic;
+  }
+
+  computeCharacteristics(characteristicsOfAvailableProducts) {
+    return characteristicsOfAvailableProducts.map((characteristics) => ({
+      display: this.$translate.instant(
+        characteristics.quantity === 1
+          ? 'ovhManagerHostingDatabaseOrderPublic_customization_characteristics_display_1'
+          : 'ovhManagerHostingDatabaseOrderPublic_customization_characteristics_display_+',
+        {
+          diskSpace: characteristics.diskSpace,
+          quantity: characteristics.quantity,
+        },
+      ),
+      characteristics,
+    }));
+  }
+
+  static buildPlanCode(diskSpace, quantity) {
+    return `sql_perso_${diskSpace}_${quantity}`;
+  }
+
+  getCharacteristicsOfAllProducts(serviceName) {
```

Comment:
I would go for something like
```suggestion
  getSpecificationOfAllProducts(serviceName) {
```
I found it more adequate for a product (if I understand well the meaning here) but it's nitpicking

Reply:
I agree but it's the way it's already called in the whole `class` so I decided to keep the same linguo :)